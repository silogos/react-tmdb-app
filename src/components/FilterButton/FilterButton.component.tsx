import { NavLink } from "react-router";
import { FilterButtonProps } from "./FilterButton.type";

function FilterButton({ filterId, title }: FilterButtonProps) {
  return (
    <NavLink
      to={filterId}
      className={({ isActive }) =>
        `px-4 py-2  text-nowrap rounded-lg ${
          isActive ? "bg-gray-50" : "bg-gray-400"
        }`
      }
    >
      {title}
    </NavLink>
  );
}

export default FilterButton;
