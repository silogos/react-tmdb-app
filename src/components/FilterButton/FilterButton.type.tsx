export type FilterButtonProps = {
  filterId: string;
  title: string;
};
