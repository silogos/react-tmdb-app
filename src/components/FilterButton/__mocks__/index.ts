export { default } from "./FilterButton.component";
