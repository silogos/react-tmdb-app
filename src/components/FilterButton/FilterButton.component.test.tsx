import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import FilterButton from "./FilterButton.component";
import { createRoutesStub } from "react-router";

describe("FilterButton Component", () => {
  const RouterStub = createRoutesStub([
    {
      path: "/popular",
      Component: () => (
        <FilterButton filterId="/popular" title="Popular Filter" />
      ),
    },
    {
      path: "/upcoming",
      Component: () => (
        <FilterButton filterId="/popular" title="Popular Filter" />
      ),
    },
  ]);

  it("displays the correct bg when active", () => {
    const component = render(<RouterStub initialEntries={["/popular"]} />);

    expect(component.getByText(/Popular Filter/)).toHaveClass("bg-gray-50");
    expect(component.container).toMatchSnapshot();
  });

  it("displays the correct bg when inactive", () => {
    const component = render(<RouterStub initialEntries={["/upcoming"]} />);

    expect(component.getByText(/Popular Filter/)).toHaveClass("bg-gray-400");
    expect(component.container).toMatchSnapshot();
  });
});
