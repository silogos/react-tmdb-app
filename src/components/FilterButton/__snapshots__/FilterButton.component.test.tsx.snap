// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterButton Component > displays the correct bg when active 1`] = `
<div>
  <a
    aria-current="page"
    class="px-4 py-2  text-nowrap rounded-lg bg-gray-50"
    data-discover="true"
    href="/popular"
  >
    Popular Filter
  </a>
</div>
`;

exports[`FilterButton Component > displays the correct bg when inactive 1`] = `
<div>
  <a
    class="px-4 py-2  text-nowrap rounded-lg bg-gray-400"
    data-discover="true"
    href="/popular"
  >
    Popular Filter
  </a>
</div>
`;
